benchmark/httpapi.py

```diff
@@ -63,14 +76,16 @@ def retrieve(self, id):
         except KeyError:
             return fail(ResultNotFound())
 
-    def query(self, filter):
+    def query(self, filter, limit):
         """
         Return matching results.
         """
-        matching = [
-            r for r in self._results.viewvalues()
-            if filter.viewitems() <= r.viewitems()
-        ]
+        matching = []
+        for result in reversed(self._sorted):
+            if filter.viewitems() <= result.viewitems():
+                matching.append(result)
+            if limit > 0 and len(matching) == limit:
```

Comment:
:grey_question: This behavior surprises me. I would have expected a 4XX error (maybe BadRequest? especially now that it's available) for negative limits and possibly just an empty response for limit of 0.

I don't have strong opinions on this as this appears to be somewhat of a corner case, so I'll leave this as optional, but we should potentially think about documenting this behavior in the interface definition.